data/blog/session-replay-sdk-bundle-size-optimizations.md

```diff
@@ -0,0 +1,208 @@
+---
+title: 'Session Replay SDK Bundle Size Optimizations'
+date: '2023-10-31'
+tags: ['sdk', 'javascript', 'replay', 'session-replay']
+draft: false
+summary: 'An in-depth analysis of how we managed to cut the Session Replay SDK bundle size by X%.'
+images: [/images/session-replay-sdk-bundle-size-optimizations/hero.jpg]
+layout: PostLayout
+canonicalUrl: https://sentry.engineering/blog/session-replay-sdk-bundle-size-optimizations
+authors: ['francesconovy']
+---
+
+[Bundle Size matters](https://blog.sentry.io/js-browser-sdk-bundle-size-matters/) - this is something we SDK engineers at Sentry are acutely aware of.
+In an ideal world, you'd get all the functionality you want with no additional bundle size - oh, would't that be nice?
+Sadly, in reality any feature we add to the JavaScript SDK results in additional bundle size for the SDK - there is always a trade off to be made.
+
+With [Session Replay](https://docs.sentry.io/product/session-replay/), this topic is especially challenging.
+Session Replay allows to capture what's going on in a users browsers, which can help developers to debug errors or problems the user is experiencing.
+While this can be incredibly helpful, there is also a considerable amount of JavaScript code required to actually make this possible - thus leading to an increased bundle size.
+
+In version 7.73.0 of the JavaScript SDKs, we updated the underlying [rrweb](https://github.com/rrweb-io/rrweb) package from v1 to v2.
```

Comment:
Since we tend to deviate a bit from upstream, lets link to our fork:

```suggestion
In version 7.73.0 of the JavaScript SDKs, we updated the underlying [rrweb](https://github.com/getsentry/rrweb) package from v1 to v2.
```